app/scrapers/courses_scraper.rb

```diff
@@ -4,54 +4,70 @@
 require 'open-uri'
 require 'nokogiri'
 
-require_relative 'scraper_common.rb'
+require_relative 'scraper_common'
 include ScraperCommon
 
-require_relative '../models/courses.rb'
+require_relative '../models/courses'
 
-prog_name = "courses_scraper"
+prog_name = 'courses_scraper'
 logger = ScraperCommon::logger
 
 # List semesters in year in testudo's format
 # 2018 -> 201801, 201805, 201808, 201812
+# @type [Array<String>]
 semesters = ScraperCommon::get_semesters(ARGV)
 
-logger.info(prog_name) {semesters}
+logger.info(prog_name) { semesters }
 
 # Get the urls for all the department pages
 dep_urls = []
 semesters.each do |semester|
-  logger.info(prog_name) {"Searching for courses in term #{semester}"}
+  logger.info(prog_name) { "Searching for courses in term #{semester}" }
 
   base_url = "https://app.testudo.umd.edu/soc/#{semester}"
 
-  Nokogiri::HTML(open(base_url)).search('span.prefix-abbrev').each do |e|
+
+  ScraperCommon::get_page(base_url, prog_name).search('span.prefix-abbrev').each do |e|
     dep_urls << "https://app.testudo.umd.edu/soc/#{semester}/#{e.text}"
   end
 
-  logger.info(prog_name) {"#{dep_urls.length} department/semesters so far"}
+  logger.info(prog_name) { "#{dep_urls.length} department/semesters so far" }
 end
 
+##
 # safely formats to UTF-8
-def utf_safe text
-  if !text.valid_encoding?
-    text = text.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
-  end
+#
+# @param [String] text
+# @return [String]
+#
+def utf_safe(text)
+  text = text.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '') unless text.valid_encoding?
   text
 end
 
 queries = []
+dept_count = Float(dep_urls.length)
 
 # add the courses from each department to the database
-dep_urls.each do |url|
-  dept_id = url.split('/soc/')[1][7,10]
-  semester = url.split('/soc/')[1][0,6]
+dep_urls.each_with_index do |url, i|
+  dept_id = url.split('/soc/')[1][7, 10]
+  semester = url.split('/soc/')[1][0, 6]
   courses = []
-  table_name = "courses"
-
-  logger.info(prog_name) {"Getting courses for #{dept_id} (#{semester})"}
+  table_name = 'courses'
+
+  progress = (Float(i) / dept_count * 100.0).round 2
```

Comment:
would use `i.to_f` here instead, seems to be the more popular choice

Reply:
sure?